File: functions/src/domain/entities/user-public.ts




export class UserPublic {

    constructor(
       public readonly uid: string,
       public readonly firstName: string,
       public readonly lastName: string,
       public agoraUid: number,
    ) {}

    get fullName () : string {
        return `${this.firstName} ${this.lastName}`;
    }

}

File: flutter_app/lib/core/domain/entities/user_public.dart
import 'package:flutter_group_chat_app_with_firebase/core/domain/services/auth_service.dart';
import 'package:flutter_group_chat_app_with_firebase/injection_container.dart';
import '../../../features/chat/domain/chat_utils.dart' as chatUtils;

class UserPublic {
  final String uid;
  final String firstName;
  final String lastName;

  UserPublic({required this.uid, required this.firstName, required this.lastName});

  String get fullName => "$firstName $lastName";

  String get conversationId {
    return chatUtils.getDirectConversationId([uid, getIt.get<AuthService>().loggedUid!]);
  }

  static UserPublic empty() => UserPublic(uid: '', firstName: '', lastName: '');
}